src/extensionProvider.ts

```diff
@@ -94,29 +94,32 @@ export class ExtensionProvider {
     return !!this.account;
   }
 
-  async signTransaction(transaction: ITransaction) {
+  async signTransaction<T extends ITransaction>(transaction: T): Promise<T> {
     await this.signTransactions([transaction]);
```

Comment:
I think you should return the await here

Reply:
Fixed.